src/coprocessor/dag/aggr_fn/impl_bit_op.rs

```diff
@@ -0,0 +1,467 @@
+            &[Some(0xffffffffffffffff as u64 as i64)]
+        );
+
+        // 7 & 4 == 4
+        state.update(&mut ctx, &Some(7i64)).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(7)]);
+
+        state.update(&mut ctx, &Some(4i64)).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(4)]);
+
+        state.update_repeat(&mut ctx, &Some(4), 10).unwrap();
+        state
+            .update_repeat(&mut ctx, &Option::<Int>::None, 7)
+            .unwrap();
+
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(4)]);
+
+        // Reset the state
+        let mut state = function.create_state();
+        // 7 & 1 == 1
+        state.update(&mut ctx, &Some(7i64)).unwrap();
+        state
+            .update_vector(&mut ctx, &[Some(1i64), None, Some(1i64)])
+            .unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(1)]);
+
+        // 7 & 1 & 2 == 0
+        state.update(&mut ctx, &Some(2i64)).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(0)]);
+    }
+
+    #[test]
+    fn test_bit_or() {
+        let mut ctx = EvalContext::default();
+        let function = AggrFnBitOp::<BitOr>(std::marker::PhantomData);
+        let mut state = function.create_state();
+
+        let mut result = [VectorValue::with_capacity(0, EvalType::Int)];
+
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(0)]);
+
+        state.update(&mut ctx, &Option::<Int>::None).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(0)]);
+
+        // 1 | 4 == 5
+        state.update(&mut ctx, &Some(1i64)).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(1)]);
+
+        state.update(&mut ctx, &Some(4i64)).unwrap();
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(5)]);
+
+        state.update_repeat(&mut ctx, &Some(8), 10).unwrap();
+        state
+            .update_repeat(&mut ctx, &Option::<Int>::None, 7)
+            .unwrap();
+
+        result[0].clear();
+        state.push_result(&mut ctx, &mut result).unwrap();
+        assert_eq!(result[0].as_int_slice(), &[Some(13)]);
+
+        // 13 | 2 == 15
+        state.update(&mut ctx, &Some(2i64)).unwrap();
+        state
+            .update_vector(&mut ctx, &[Some(2i64), None, Some(1i64)])
```

Comment:
It does not do the same thing as the comment ("13 | 2 == 15") says...

Reply:
@sticnarf I think there is no need to comment above `update_vector`.